DevCycle/SSEConnection.swift

```diff
@@ -11,21 +11,55 @@ import LDSwiftEventSource
 
 public typealias MessageHandler = (String) -> Void
 
-class SSEConnection {
+protocol SSEConnectionProtocol {
+    var connected: Bool { get }
+    func openConnection()
+    func close()
+    func reopen()
+}
+
+class SSEConnection: SSEConnectionProtocol {
     private var connection: EventSource
+    private var url: URL
+    private var handler: MessageHandler
+    
+    var connected: Bool
     
     init(url: URL, eventHandler: @escaping MessageHandler) {
+        self.url = url
+        self.handler = eventHandler
+        self.connected = false
+        self.connection = SSEConnection.makeConnection(url: url, eventHandler: eventHandler)
+        self.openConnection()
+    }
+    
+    public func openConnection() {
         Log.debug("Establishing realtime streaming connection.")
-        let handler = Handler(handler: eventHandler)
-        self.connection = EventSource(config: EventSource.Config(
-            handler: handler,
+        self.connection = SSEConnection.makeConnection(url: self.url, eventHandler: self.handler)
```

Comment:
you could also move this one to the `reopen` method so that you aren't making two of them on construction